Tests/BSONTests/DocumentTests.swift

```diff
@@ -345,249 +296,121 @@ final class DocumentTests: MongoSwiftTestCase {
         expect(doc["a2"]).to(equal(arr2))
     }
 
-    // exclude Int64 value on 32-bit platforms
-    static let overwritables: Document = [
-        "double": 2.5,
-        "int32": .int32(32),
-        "int64": .int64(Int64.max),
-        "bool": false,
-        "decimal": .decimal128(Decimal128("1.2E+10")!),
-        "oid": .objectId(ObjectId()),
-        "timestamp": .timestamp(Timestamp(timestamp: 1, inc: 2)),
-        "datetime": .datetime(Date(msSinceEpoch: 1000))
-    ]
-
-    static let nonOverwritables: Document = [
-        "string": "hello",
-        "nil": .null,
-        "doc": ["x": 1],
-        "arr": [1, 2]
-    ]
-
-    // test replacing `Overwritable` types with values of their own type
-    func testOverwritable() throws {
-        // make a deep copy so we start off with uniquely referenced storage
-        var doc = DocumentTests.overwritables.withBSONPointer { ptr in
-            Document(copying: ptr)
-        }
-
-        // save a reference to original bson_t so we can verify it doesn't change
-        let pointer = doc.pointerAddress
+    // Test overwriting with value of same type
+    func testOverwritingValues() throws {
```

Comment:
previously we followed a separate code path when overwriting values of fixed lengths (e.g. ObjectID) because libbson had special API to do that in-place. we don't do that now (though we could revisit it as a future optimization) so I've just turned the tests for all that overwriting-in-place behavior into general tests for overwriting behavior since that still seemed useful to have.